### ISSUE TITLE
Compiler warning with Elixir 1.18.0

### ISSUE THREAD

**Issue by nathany-copia** [bitwalker/timex#769]
### Steps to reproduce

mix compile

### Description of issue

- What are the expected results?

no compiler warning

- If you are certain this is a bug in Timex, do you have references I

```
warning: Range.new/2 and first..last default to a step of -1 when last < first. Use Range.new(first, last, -1) or first..last//-1, or pass 1 if that was your intention
  (timex 3.7.11) lib/parse/datetime/helpers.ex:168: Timex.Parse.DateTime.Helpers.integer/1
  (timex 3.7.11) lib/parse/datetime/parsers.ex:24: Timex.Parse.DateTime.Parsers.year4/1
  (timex 3.7.11) lib/parse/datetime/tokenizers/directive.ex:110: Timex.Parse.DateTime.Tokenizers.Directive.get/5
  (combine 0.10.0) lib/combine/parsers/base.ex:154: Combine.Parsers.Base.pipe_impl/3
  (combine 0.10.0) lib/combine/parsers/base.ex:490: Combine.Parsers.Base.preserve_ignored_impl/2
```

**Comment by flmn** [comment]
and this:

warning: Module.eval_quoted/4 is deprecated. Use Code.eval_quoted/3 instead
      │
 1205 │     Module.eval_quoted(__MODULE__, day_quoted, [], __ENV__)
      │            ~
      │
      └─ lib/timex.ex:1205:12: Timex (module)

**Comment by nathany-copia** [comment]
#767

**Comment by bitwalker** [comment]
Fixed via #767

**Comment by nathany-copia** [comment]
Thanks @bitwalker.

@flmn #767 fixed the issue I reported, but you may want to open a separate issue for what you reported.

#759 is still causing warnings too -- when I compile with latest.

`{:timex, "~> 3.7", github: "bitwalker/timex", ref: "cc649c7a5", override: true},`


```
==> timex
Compiling 62 files (.ex)
warning: defining a Gettext backend by calling

    use Gettext, otp_app: ...

is deprecated. To define a backend, call:

    use Gettext.Backend, otp_app: :my_app

Then, instead of importing your backend, call this in your module:

    use Gettext, backend: MyApp.Gettext

  lib/l10n/gettext.ex:2: Timex.Gettext (module)

warning: min..max inside match is deprecated, you must always match on the step: min..max//var or min..max//_ if you want to ignore it
  lib/format/datetime/formatter.ex:893: Timex.Format.DateTime.Formatter.width_spec/1

      warning: Module.eval_quoted/4 is deprecated. Use Code.eval_quoted/3 instead
      │
 1205 │     Module.eval_quoted(__MODULE__, day_quoted, [], __ENV__)
      │            ~
      │
      └─ lib/timex.ex:1205:12: Timex (module)

      warning: Module.eval_quoted/4 is deprecated. Use Code.eval_quoted/3 instead
      │
 1303 │     Module.eval_quoted(__MODULE__, month_quoted, [], __ENV__)
      │            ~
      │
      └─ lib/timex.ex:1303:12: Timex (module)

    warning: Module.eval_quoted/4 is deprecated. Use Code.eval_quoted/3 instead
    │
 22 │     Module.eval_quoted(__MODULE__, quoted, [], __ENV__)
    │            ~
    │
    └─ lib/timezone/utils.ex:22:12: Timex.Timezone.Utils (module)

    warning: Module.eval_quoted/4 is deprecated. Use Code.eval_quoted/3 instead
    │
 42 │     Module.eval_quoted(__MODULE__, quoted, [], __ENV__)
    │            ~
    │
    └─ lib/timezone/utils.ex:42:12: Timex.Timezone.Utils (module)

```
